uap_tracker/frame_processor.py

```diff
@@ -186,7 +188,8 @@ def keypoints_from_bg_subtraction(self, frame_grey, stream):
             frame_masked_background = cv2.bitwise_and(frame_grey, frame_grey, mask=foreground_mask)
 
             # Detect new objects of interest to pass to tracker
-            key_points = utils.perform_blob_detection(frame_masked_background, self.detection_sensitivity)
+            # key_points = utils.perform_blob_detection(frame_masked_background, self.detection_sensitivity)
+            key_points = self.blob_detector.detectBB(frame_masked_background)
```

Comment:
One thing I would say here is that the BB suffix of the method makes me think you are expecting a bounding box as a return type versus key points.